languages/scxml/src/main/java/de/jplag/scxml/parser/model/executable_content/ExecutableContent.java

```diff
@@ -0,0 +1,64 @@
+package de.jplag.scxml.parser.model.executable_content;
+
+import static de.jplag.scxml.parser.model.executable_content.SimpleExecutableContent.Type.*;
+
+import java.util.Set;
+
+import org.w3c.dom.Node;
+
+import de.jplag.scxml.parser.model.StatechartElement;
+import de.jplag.scxml.parser.util.NodeUtil;
+
+/**
+ * Represents executable content in an SCXML statechart, which are elements that can be executed
+ * during state transitions, state entry, state exit or in conditional statements.
+ */
+public interface ExecutableContent extends StatechartElement {
+
+    /**
+     * Defines the set of allowed XML element names that are considered valid executable content. {@literal <else>} and
+     * {@literal <elseif>} elements are not allowed as they may only present as children of an {@literal <if>} element.
+     */
+    Set<String> ALLOWED_XML_ELEMENTS = Set.of("raise", "if", "foreach", "log", "assign", "script", "send", "cancel");
+
+    /**
+     * String constant for the <else> element.
+     */
+    String ELSE_ELEMENT = "else";
+
+    /**
+     * String constant for the "event" attribute.
+     */
+    String EVENT_ATTRIBUTE = "event";
+
+    /**
+     * String constant for the "sendid" attribute.
+     */
+    String SEND_ID_ATTRIBUTE = "sendid";
+
+    /**
+     * String constant for the "delay" attribute.
+     */
+    String DELAY_ATTRIBUTE = "delay";
+
+    /**
+     * Constructs a concrete instance of ExecutableContent based on the name of the given node.
+     * @param node the node to create the ExecutableContent from
+     * @return the constructed ExecutableContent
+     * @throws IllegalArgumentException if the node name is not allowed or the executable content could not be created
+     */
+    static ExecutableContent fromNode(Node node) throws IllegalArgumentException {
+        return switch (node.getNodeName()) {
+            case "if" -> If.fromNode(node);
```

Comment:
Are there no constants defined for these String literals? e.g. in Node?

Reply:
No, as those are SCXML-specific. I can define them in `ExecutableContent`, if you prefer that though.

Reply:
Yes, that would be better!